nvflare/app_common/widgets/log_streaming.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import threading
+from builtins import dict as StreamContext
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_constant import FLContextKey, ReturnCode, SystemComponents
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.workspace import Workspace
+from nvflare.app_common.streamers.file_streamer import FileStreamer
+from nvflare.widgets.widget import Widget
+
+LOG_STREAM_EVENT_TYPE = "stream_log"
+
+
+class LogConst(object):
+    CLIENT_NAME = "client_name"
+    JOB_ID = "job_id"
+    LOG_DATA = "log_data"
+
+
+class LogSender(Widget):
```

Comment:
since this component only works in client, what happen if users put this in server side?
Do we need some checking against that?

Reply:
You can do this now by checking the fl_ctx.get_process_type() to be ProcessType.CLIENT_PARENT.
Do this in handle_event.